sea-orm-macros/src/derives/entity_model.rs

```diff
@@ -58,6 +59,18 @@ pub fn expand_derive_entity_model(data: Data, attrs: Vec<Attribute>) -> syn::Res
     let mut primary_keys: Punctuated<_, Comma> = Punctuated::new();
     let mut primary_key_types: Punctuated<_, Comma> = Punctuated::new();
     let mut auto_increment = true;
+
+    #[cfg(feature = "with-table-iden")]
+    if let Some(table_name) = table_name {
+        let table_field_name = Ident::new("Table", Span::call_site());
+        columns_enum.push(quote! {
+            #[sea_orm(table_name=#table_name)]
+            #[strum(disabled)]
+            #table_field_name
+        });
+        columns_trait
+            .push(quote! { Self::#table_field_name => panic!("Table cannot be used as a column") });
```

Comment:
I was not sure what to do with that, I figured it was best to abort early

Reply:
Make sense to me